doseCalc/+DoseEngines/matRad_ParticlePencilBeamEngineAbstract.m

```diff
@@ -0,0 +1,932 @@
+                    yyaxis left;
+                    plot(this.machine.data(energyIx).LatCutOff.depths,this.machine.data(energyIx).LatCutOff.CutOff,'LineWidth',2),hold on
+                    plot(this.machine.data(energyIx).depths,(this.machine.data(energyIx).sigma1),':','LineWidth',2),grid on,hold on,ylabel('mm')
+                    yyaxis right;
+                    plot(this.machine.data(energyIx).depths,(this.machine.data(energyIx).sigma2),'-.','LineWidth',2),grid on,hold on,ylabel('mm')
+                    legend({'Cutoff','sigma1','sigma2'});
+                else
+                    yyaxis left;plot(this.machine.data(energyIx).LatCutOff.depths,this.machine.data(energyIx).LatCutOff.CutOff,'LineWidth',2),hold on,ylabel('mm')
+                    yyaxis right;subplot(313),plot(this.machine.data(energyIx).depths,this.machine.data(energyIx).sigma,'LineWidth',2),grid on,hold on
+                    legend({'Cutoff','sigma'});ylabel('mm')
+                end
+
+                set(gca,'FontSize',12),xlabel('z [mm]'),  ylabel('mm')
+
+                % plot cutoff of different energies
+                figure,set(gcf,'Color',[1 1 1]);
+                cnt = 1;
+                for i = vEnergiesIx
+                    plot(this.machine.data(i).LatCutOff.depths,this.machine.data(i).LatCutOff.CutOff,'LineWidth',1.5),hold on
+                    cellLegend{cnt} = [num2str(this.machine.data(i).energy) ' MeV'];
+                    cnt = cnt + 1;
+                end
+                grid on, grid minor,xlabel('depth in [mm]'),ylabel('lateral cutoff in [mm]')
+                title(['cutoff level = ' num2str(cutOffLevel)]),
+                ylim = get(gca,'Ylim');    set(gca,'Ylim',[0 ylim(2)+3]),    legend(cellLegend)
+            end
+        end
+    
+        function ray = initRay(this,beam,j)           
+            ray = initRay@DoseEngines.matRad_PencilBeamEngineAbstract(this,beam,j);
+
+            % calculate initial sigma for all bixel on current ray
+            ray.sigmaIni = matRad_calcSigmaIni(this.machine.data,ray,ray.SSD);
+            
+            % Since matRad's ray cast starts at the skin and base data
+            % is generated at soume source to phantom distance
+            % we can explicitly correct for the nozzle to air WEPL in
+            % the current case.
+            if this.airOffsetCorrection
+                nozzleToSkin = (ray.SSD + this.machine.meta.BAMStoIsoDist) - this.machine.meta.SAD;
+                ray.radDepthOffset = 0.0011 * (nozzleToSkin - this.machine.meta.fitAirOffset);
+            else
+                ray.radDepthOffset = 0;
+            end
+
+            % just use tissue classes of voxels found by ray tracer
+            if this.calcBioDose
+                ray.vTissueIndex = this.vTissueIndex(ray.subIxVdoseGrid,:);
+                ray.vAlphaX = this.vAlphaX(ray.subIxVdoseGrid);
+                ray.vBetaX  = this.vBetaX(ray.subIxVdoseGrid);
```

Comment:
I think there is an inconsistency here.
this.vAlphaX and the other arrays have the dimension (#VoxelsOnDoseGrid,1) while ray.subIxVdoseGrid contains indexes referred to this.VdoseGrid which is smaller and only contains the voxels on which the dose calculation is actually performed.

Reply:
I fixed this by masking them when they are created and tested correct assignment:
https://github.com/e0404/matRad/blob/428bbedc0acc9e1b0be3df1471bca19995f3849c/doseCalc/%2BDoseEngines/matRad_ParticlePencilBeamEngineAbstract.m#L420-L423